model/facade/action/PartialUpdateTasksAction.php

```diff
@@ -127,6 +127,20 @@ protected function doExecute() {
                 $discardedTasks[] = $task;
                 unset($this->tasks[$i]);
             }
+
+            // Check if init was updated and endTime should be converted to 24:00
+            // (case it was a 0-hour task and it's not anymore)
+            $currentEndTime = $oldTask->getEnd();
+            $newInitTime = $this->tasks[$i]->getInit();
+            if ($newInitTime > $currentEndTime && $currentEndTime === 0) {
+                $this->tasks[$i]->setEnd(1440);
+            }
+
+            // Check if init was updated and endTime should be converted to 0h
+            // (case it was a regular task and now it's a 0-hour task)
+            if ($newInitTime === 0 && $currentEndTime == 1440) {
+                $this->tasks[$i]->setEnd(0);
+            }
```

Comment:
It's possible to call the update operation with no init time set, meaning that you don't want to update that field.

In that case, `$newInitTime` would be null. It looks like the code works properly because null is not bigger than $currentEndDate nor it equallls (===) zero, but I would wrap all this block in `if $task->isInitTimeDirty() ...`. I would find it easier to understand what's going on and prevent exotic comparisons.

Reply:
That makes sense, I've just updated the last commit with this suggestion